Cannot delete project - PostgreSQL only supports 65535 parameters

When attempting to delete a seemingly empty project in harbor's UI an 'Internal Server Error' message is displayed and the harbor core pod logs show;

> 2021-04-16T15:11:31Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"unknown: pq: got 98119 parameters but PostgreSQL only supports 65535 parameters"}]}


We are using harbor 2.2.0 deployed on a Kubernetes cluster, with an external Postgres 9.6.3.

The s3 backend and database has many projects which have been deleted from the UI but remain in the DB/Storage and may be causing some issues, but without knowing what the query is that fails I can't debug any further.

The API call is not that useful;

```
curl -X DELETE "https://harbor.REDACTED.TLD/api/v2.0/projects/vanilla" -H  "accept: application/json" -H  "X-Is-Resource-Name: false" -H  "authorization: Basic REDACTED=" -v
...
< HTTP/2 500
< server: nginx/1.19.0
< date: Mon, 19 Apr 2021 18:35:29 GMT
< content-type: application/json; charset=utf-8
< content-length: 66
< set-cookie: sid=4f9f861b320bbf8a18634c29bf35c7df; Path=/; HttpOnly
< x-request-id: 38b16e3a4f2a92c74337c32fca589e4c
< strict-transport-security: max-age=15724800; includeSubDomains
<
{"errors":[{"code":"UNKNOWN","message":"internal server error"}]}
```

The logs from the core service are literally all i've shown you already, but here are some lines around just for clarity;

```
2021-04-19T18:34:50Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with Handler 'ArtifactWebhook' on topic 'PULL_ARTIFACT': ID-15597, Repository-docker.io/jaegertracing/jaeger-agent Tags-[latest] Digest-sha256:83219dff5bdebc835714e9f0bdc7710058056e69e2ff1987e182baaa557083cc Operator- OccurAt-2021-04-19 18:34:50
2021-04-19T18:35:29Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"unknown: pq: got 98119 parameters but PostgreSQL only supports 65535 parameters"}]}
2021-04-19T18:36:20Z [INFO] [/pkg/notifier/notifier.go:204]: Handle notification with Handler 'InternalArtifact' on topic 'PULL_ARTIFACT': ID-2091, Repository-docker.io/gitlab/gitlab-runner-helper Tags-[x86_64-a8a019e0] Digest-sha256:5e1c83270146c0cb0b26708f1c4f006de3c298e71b036acdcd502eeb87fb0b24 Operator- OccurAt-2021-04-19 18:36:20
```

--- a/src/pkg/robot/dao/dao.go
+++ b/src/pkg/robot/dao/dao.go
@@ -2,11 +2,12 @@ package dao
 
 import (
 	"context"
+	"time"
+
 	"github.com/goharbor/harbor/src/lib/errors"
 	"github.com/goharbor/harbor/src/lib/orm"
 	"github.com/goharbor/harbor/src/lib/q"
 	"github.com/goharbor/harbor/src/pkg/robot/model"
-	"time"
 )
 
 // DAO defines the interface to access the robot data model
@@ -121,14 +122,12 @@ func (d *dao) List(ctx context.Context, query *q.Query) ([]*model.Robot, error) 
 }
 
 func (d *dao) DeleteByProjectID(ctx context.Context, projectID int64) error {
-	qs, err := orm.QuerySetter(ctx, &model.Robot{}, &q.Query{
-		Keywords: map[string]interface{}{
-			"project_id": projectID,
-		},
-	})
+	ormer, err := orm.FromContext(ctx)
 	if err != nil {
 		return err
 	}
-	_, err = qs.Delete()
+
+	_, err = ormer.Raw("DELETE FROM robot WHERE project_id = ?", projectID).Exec()
+
 	return err
 }

